src/cgls/cgls.c

```diff
@@ -36,6 +37,8 @@ static char **arg_names = NULL;
 static int arg_full = -1;
 static const char* arg_machine = NULL;
 
+STATIC_DESTRUCTOR_REGISTER(arg_names, freep);
```

Comment:
should retain the comment the free() call mentioned before: `/* don't free the strings */`

Reply:
Updated.